src/main/java/org/axonframework/springboot/nativex/AxonRuntimeHints.java

```diff
@@ -0,0 +1,238 @@
+        axonClasses.stream()
+                   .flatMap(c -> Arrays.stream(c.getMethods()))
+                   .filter(this::isAxonMethod)
+                   .forEach(m -> {
+                       if (m.getParameterTypes().length > 0) {
+                           apiClasses.add(m.getParameterTypes()[0]);
+                       }
+                       if (m.isAnnotationPresent(QueryHandler.class)) {
+                           Class<?> returnType = m.getReturnType();
+                           if (!Collection.class.isAssignableFrom(returnType)) {
+                               apiClasses.add(returnType);
+                           }
+                       }
+                   });
+    }
+
+    private Set<Class<?>> getClasses(ClassLoader classLoader, String packageName) {
+        InputStream stream = classLoader.getResourceAsStream(
+                packageName.replaceAll("[" + PACKAGE_SEPARATOR + "]", FOLDER_SEPARATOR)
+        );
+        assert stream != null;
+        BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+        List<String> lines = reader.lines().toList();
+
+        Stream<Class<?>> current = lines.stream()
+                                        .filter(this::isClassFile)
+                                        .map(line -> getClass(line, packageName));
+
+        Stream<Class<?>> nested = lines.stream()
+                                       .filter(this::isPackageFolder)
+                                       .map(String::trim)
+                                       .map(child -> setChildPackageName(packageName, child))
+                                       .map(pName -> getClasses(classLoader, pName))
+                                       .flatMap(Set::stream);
+
+        return Stream.concat(current, nested).collect(Collectors.toSet());
+    }
+
+    private Class<?> getClass(String className, String packageName) {
+        try {
+            return Class.forName(
+                    packageName + PACKAGE_SEPARATOR + className.substring(0, className.lastIndexOf(PACKAGE_SEPARATOR)));
+        } catch (ClassNotFoundException e) {
+            return null;
+        }
+    }
+
+    private boolean isClassFile(String path) {
+        return path.endsWith(".class");
+    }
+
+    private boolean isPackageFolder(String path) {
+        return !isClassFile(path);
+    }
+
+    private String setChildPackageName(String parent, String child) {
+        return parent + PACKAGE_SEPARATOR + child;
+    }
+
+    private boolean isAxonClass(Class<?> clazz) {
+        return clazz.isAnnotationPresent(Aggregate.class) ||
+                clazz.isAnnotationPresent(ProcessingGroup.class) ||
+                clazz.isAnnotationPresent(Saga.class) ||
+                Arrays.stream(clazz.getMethods()).anyMatch(this::isAxonMethod);
+    }
+
+    private boolean isAxonMethod(Method method) {
+        return method.isAnnotationPresent(EventHandler.class) ||
```

Comment:
With a similar point as under `isAxonClass`, I am curious whether this also works if somebody has meta-annotated their annotation with ours.

Furthermore, I would assume we are missing some method annotations.
The `AxonNativeHints` deliberately adds things like `@StartHandler`, `@ShutdownHandler`. and `@MessageHandler`, for example.
I would guess those need to be included here too.

Reply:
If we check for `@MessageHandler` using `AnnotationUtils.isAnnotationPresent()` , then we don't need to check separately for `QueryHandler`, `CommandHandler`, etc.

Reply:
Typically, Axon is triggered using annotation attributes, which are indicated by the `@HasHandlerAttributes` annotation. If that annotation is available on a method or constructor, it needs to be available for reflection by Axon.

Reply:
Changed for check on `@MessageHandler`